src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala

```diff
@@ -69,11 +94,14 @@ class ClusterMonitorSupervisor(monitorConfig: MonitorConfig, dataprocConfig: Dat
     context.actorOf(ClusterMonitorActor.props(cluster, monitorConfig, dataprocConfig, gdDAO, googleComputeDAO, googleIamDAO, googleStorageDAO, dbRef, clusterDnsCache, authProvider))
   }
 
-  def startClusterMonitorActor(cluster: Cluster, recreate: Boolean = false): Unit = {
+  def startClusterMonitorActor(cluster: Cluster, watchMessageOpt: Option[ClusterSupervisorMessage] = None): Unit = {
     val child = createChildActor(cluster)
 
-    if (recreate && monitorConfig.recreateCluster) {
-      context.watchWith(child, RecreateCluster(cluster))
+    watchMessageOpt.foreach {
+      case RecreateCluster(_) if !monitorConfig.recreateCluster =>
+        // don't recreate clusters if not configured to do so
+      case watchMsg =>
+        context.watchWith(child, watchMsg)
```

Comment:
For the `stopAfterCreate` case, does this spin off an actor that monitors the stopping or?

Reply:
Basically it installs a "monitor" on the child actor. When the child actor is terminated, the `watchMsg` (`StopClusterAfterCreation` in this example) will be sent to the supervisor actor, causing the cluster to then be stopped.